### ISSUE TITLE
Implement Python bindings for BallistaContext

### ISSUE THREAD

**Issue by andygrove** [apache/arrow-datafusion#861]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
We have Python bindings for DataFusion's ExecutionContext. It would be good to also support Ballista's BallistaContext so that we can use Python to run distributed queries.

**Describe the solution you'd like**
Probably something like this?

```python
import ballista

ctx = ballista.BallistaContext
df = ctx.read_parquet(...)
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by djKooks** [comment]
@andygrove hope to work on this, if you don't have any other plan.

**Comment by andygrove** [comment]
Thank you @djKooks that would be great

**Comment by djKooks** [comment]
@andygrove 
Would it be okay to put like following?
```
...
ballista/
   - rust/
   - ui/
   - python/     <- create binding here
datafusion/
datafusion-cli/
...
```

**Comment by andygrove** [comment]
Yes, I think that makes sense.

**Comment by andygrove** [comment]
I would be interested to hear what others think though. @alamb @Dandandan @jorgecarleitao @houqp do you have an opinion on this?

**Comment by jorgecarleitao** [comment]
I agree that this makes the most sense 👍

Out of curiosity, do the bindings come with the client and executors, or just the client?

**Comment by alamb** [comment]
I agree having a separate python binding for Ballista in the location suggested by @djKooks  in https://github.com/apache/arrow-datafusion/issues/861#issuecomment-899453613 makes sense to me

**Comment by djKooks** [comment]
@alamb @jorgecarleitao @andygrove thanks for suggestion 🙇 

> do the bindings come with the client and executors, or just the client?

I think it will be enough to do with client only in first step, but do you have any more suggestion?

**Comment by alamb** [comment]
I do not have any more to add here -- since I don't use the python bindings myself I don't have a lot to offer with specifics

**Comment by djKooks** [comment]
@andygrove  @alamb @jorgecarleitao thanks for comment.
Will start implementation in following branch https://github.com/apache/arrow-datafusion/pull/988
(will request for review when ready 🙇 )
